ext/cdi/jersey-cdi1x-validation/src/main/java/org/glassfish/jersey/ext/cdi1x/validation/internal/CdiInterceptorWrapperExtension.java

```diff
@@ -171,4 +176,12 @@ public void destroy(ValidationInterceptor instance,
             }
         });
     }
+
+    /* package */ Cache<Class<?>, Boolean> getJaxRsResourceCache() {
+        /* CDI injection hack */
+        if (jaxRsResourceCache == null) {
```

Comment:
Should we synchronize this or are we sure only one thread can invoke it the first time?. Something like:
```
    /* package */ Cache<Class<?>, Boolean> getJaxRsResourceCache() {
        /* CDI injection hack */
        if (jaxRsResourceCache == null) {
           synchronize(this) {
                if (jaxRsResourceCache == null) {
                      jaxRsResourceCache = new Cache<>(clazz -> Resource.from(clazz) != null);
                }
           }
        }
        return jaxRsResourceCache;
    }
```

Reply:
For the sake, it better be initialized in a constructor of the `CdiInterceptorWrapper` to be called by just a single thread.

Reply:
I don't know all the details about how is going to be used, but imagine that there is a singleton that invokes that method and it saves it as a field. So next time it wants to use the cache, it read it from the field (instead of getting it from getJaxRsResourceCache()).

It will not harm to have that lock because it will only lock at the beginning and we get rid of possible memory issues.

Reply:
It happens that all the `if (jaxRsResourceCache == null) {` block is unnecessary, so we could have dropped it